src/nfa/nfa-intersection.cc

```diff
@@ -319,6 +320,20 @@ class Intersection
         }
     }
 
+    // Alternative for the above that uses SynchronizedUniverzalIterator
+    void compute_for_state_pair_using_sui()
+    {
+        Mata::Util::SynchronizedUniverzalIterator<TransSymbolStates> sui(2);
+        sui.push_back(lhs.transitionrelation[pair_to_process.first]);
+        sui.push_back(rhs.transitionrelation[pair_to_process.second]);
+
+        while (sui.advance()){
+            std::vector<Mata::Util::OrdVector<TransSymbolStates>::const_iterator> moves = sui.get_current();
+            assert(moves.size() == 2);
```

Comment:
I would advise to add string comment to the assertion, so one can know what the failed assertion means.
```assert(moves.size() == 2 && "Exactly two ... must be returned")```

Reply:
ok, if I find something to say that does not just repeat what the assert says